### PR TITLE
add doxygen configuration

### PR DESCRIPTION
**Pull request by makelinux**
doxygen for auto generated documentation.

Preview:

image

image

### CODE REVIEW

**Review by russellb — CHANGES_REQUESTED** [review]
How about a Makefile that makes it easy to run the generation?

Also, how did you decide on Doxygen vs alternatives? My impression had been Sphinx is kind of the standard for Python.

**Comment by makelinux** [comment]
To generate need just run `doxygen`. Let's take a look and compare efforts and results.

**Comment by makelinux** [comment]
Doxygen generates documentation for API and internals, including call graphs.

Sphinx AFAIK generates only API documentation.

File: Doxyfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+PROJECT_NAME           = "Instruct Lab CLI"
```
Comment (nathan-weinberg) [inline Doxyfile:1]:
```suggestion
PROJECT_NAME           = "InstructLab CLI"
```

**Review by nathan-weinberg — CHANGES_REQUESTED** [review]
Where do the generated docs get saved? Is there any docs anywhere as to how to install the utility?

**Comment by makelinux** [comment]
> Where do the generated docs get saved? Is there any docs anywhere as to how to install the utility?

Generated documentation is saved by default to directory html.

About Doxygen: https://www.doxygen.nl/download.html

**Comment by nathan-weinberg** [comment]
If we decide to include this, we should include docs about installing/using it - otherwise this is just an unexplained config file

**Comment by makelinux** [comment]
Yes. it can live in docs. Thank you.

**Comment by nathan-weinberg** [comment]
Putting a hold on this until the docs around it I requested are added

**Review by hickeyma — COMMENTED** [review]
Preference for sphinx as well

**Comment by leseb** [comment]
What's the latest on this? Are we aiming for Sphinx only? Thanks!

**Comment by makelinux** [comment]
Doxygen has several unique features. Just two of them: graphs, annotated directory tree.

**Comment by tiran** [comment]
Who is the target user for the documentation and how does any of these features benefit the target user?

**Comment by makelinux** [comment]
Our target audience includes not only users of the ilab CLI but also developers contributing to the project. Developers, in particular, benefit significantly from Doxygen features, as it facilitates comfortable and efficient project navigation.

**Review by hickeyma — COMMENTED** [review]
@makelinux @tiran This PR and #1184 seem to be duplicates in functionality to add infrastructure to build documentation from code. Do you mind working out which implementation we are going to progress with?

**Pull request closed without merging**